tests/unit/commands/serve-test.js

```diff
@@ -54,7 +68,159 @@ describe('serve command', function() {
+      //
+      // 1) serve command setting --live-reload-port to a taken port on a different interface:
+
+      //     AssertionError: expected 49152 to not deeply equal 49152
+      //     + expected - actual
+
+
+      //     at tests/unit/commands/serve-test.js:125:48
+      //     at lib$rsvp$$internal$$tryCatch (node_modules/rsvp/dist/rsvp.js:493:16)
+      //     at lib$rsvp$$internal$$invokeCallback (node_modules/rsvp/dist/rsvp.js:505:17)
+      //     at lib$rsvp$$internal$$publish (node_modules/rsvp/dist/rsvp.js:476:11)
+      //     at lib$rsvp$asap$$flush (node_modules/rsvp/dist/rsvp.js:1198:9)
+
+      function testVariation(variation) {
+        return testServer(variation.inputOpts, function(opts, server) {
+
+          return command.validateAndRun([
+            '--port', '0',
+            '--live-reload-port', opts.port
+          ]).then(function() {
+            var serveRun = tasks.Serve.prototype.run;
+            var runOps = serveRun.calledWith[0][0];
+
+            // sanity check. does our server meet our expectations?
+            // ubuntu 14 starts port searching around 30000, mac at 49152
+            if (variation.inputOpts.port === '0') {
+              expect(server.address().port).to.be.within(30000, 65535, 'assigning to port 0 generated an open port '+ server.address().port);
+              expect(server.address().family).to.eql(variation.expectedOpts.family);
+              expect(server.address().address).to.eql(variation.expectedOpts.address);
+            } else {
+              expect(server.address()).to.deep.eq(variation.expectedOpts, 'sanity check: test server has expected address()');
+            }
+
+            // does our live reload port meet our expectations
+            expect(runOps.liveReloadPort).to.be.within(49152, 65535, 'has live reload port '+runOps.liveReloadPort);
+            expect(server.address().port).to.not.eql(runOps.liveReloadPort);
+            expect(server.address().address).to.not.eql(runOps.liveReloadHost);
+          });
+        });
+      }
+
+      var variations = [];
+      var portZeroTries = 10;
+      while(portZeroTries > 0) {
+        variations.push({inputOpts: { port: '0', host: '::1' }, expectedOpts: { port: 0, family: 'IPv6', address: '::1' }});
+        variations.push({inputOpts: { port: '0', host: '0.0.0.0' }, expectedOpts: { port: 0, family: 'IPv4', address: '0.0.0.0' }});
+        variations.push({inputOpts: { port: '0', host: '127.0.0.1' }, expectedOpts: { port: 0, family: 'IPv4', address: '127.0.0.1' }});
+        portZeroTries--;
+      }
+
+      return Promise.all(variations.map(function(variation) {
+        return testVariation(variation);
+      })).then(function() {
+        expect(tasks.Serve.prototype.run.called).to.equal(variations.length, 'expected run to be called '+variations.length);
+      });
+    });
+  }
+
+  // will eventually fail if you set the number of iterations high enough
```

Comment:
no longer true, at least i cant make it fail anymore, which is good. i think async.series may have fixed it (instead of Promise.all)